### PR TITLE
feat(cli,core,shared): install demo connectors only in cloud

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

install demo connectors only in cloud

## Testing

Local tested

## Checklist

- [ ] `.changset-staged`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

**Review by darcyYe — APPROVED** [review]

File: packages/cli/src/commands/connector/add.ts [hunk 1]
```diff
@@ -24,12 +25,19 @@ const add: CommandModule<
         describe:
           'Add all official connectors.\n' +
           "If it's true, the specified package names will be ignored.",
+      })
+      .option('cloud', {
+        describe: 'Add additional connectors for cloud',
```
Comment (gao-sun) [inline packages/cli/src/commands/connector/add.ts:30]:
```suggestion
        describe: 'Add additional connectors for Logto Cloud',
```

File: packages/cli/src/commands/connector/utils.ts [hunk 2]
```diff
@@ -232,16 +232,21 @@ const fetchOfficialConnectorList = async () => {
   return packages;
 };
 
-export const addOfficialConnectors = async (instancePath: string) => {
+export const addOfficialConnectors = async (
+  instancePath: string,
+  excludePackageNames: string[] = []
```
Comment (gao-sun) [inline packages/cli/src/commands/connector/utils.ts:237]: maybe we can simply put a boolean here, and pass into `fetchOfficialConnectorList`, then update `packages/cli/src/commands/connector/utils.ts:220` to conditionally add `'cloud'` using `conditionalArray()`

image

File: packages/cli/src/commands/connector/utils.ts [hunk 3]
```diff
@@ -232,16 +232,21 @@ const fetchOfficialConnectorList = async () => {
...
 
-  log.info(`Found ${packages.length} official connectors`);
+  const filterdPackages = packages.filter(({ name }) => !excludePackageNames.includes(name));
+
+  log.info(`Found ${filterdPackages.length} official connectors`);
 
   await addConnectors(
     instancePath,
-    packages.map(({ name }) => name)
+    filterdPackages.map(({ name }) => name)
```
Comment (gao-sun) [inline packages/cli/src/commands/connector/utils.ts:250]: typo btw

File: packages/cli/src/commands/connector/add.ts [hunk 4]
```diff
@@ -24,12 +25,19 @@ const add: CommandModule<
...
+        type: 'boolean',
+        default: false,
+        hidden: true,
       }),
-  handler: async ({ packages: packageNames, path, official }) => {
+  handler: async ({ packages: packageNames, path, official, cloud }) => {
     const instancePath = await inquireInstancePath(path);
 
     if (official) {
```
Comment (gao-sun) [inline packages/cli/src/commands/connector/add.ts:38]: also it'll be great to throw an error if `--cloud` presents without `--official`

File: packages/cli/src/commands/connector/utils.ts (hunk 2 above)
Comment (wangsijie) [reply packages/cli/src/commands/connector/utils.ts]: updated

File: packages/cli/src/commands/connector/add.ts (hunk 4 above)
Comment (wangsijie) [reply packages/cli/src/commands/connector/add.ts:41]: updated

**Pull request merged**